src/compile/data/optimize.ts

```diff
@@ -112,8 +120,13 @@ function getLeaves(roots: DataFlowNode[]) {
 /**
  * Inserts an Intermediate ParseNode containing all non-conflicting Parse fields and removes the empty ParseNodes
  */
-export function mergeParse(node: DataFlowNode) {
-  const parseChildren = node.children.filter((x): x is ParseNode => x instanceof ParseNode);
+export function mergeParse(node: DataFlowNode): optimizers.OptimizerFlags {
```

Comment:
I wonder if it would make more sense to have this function merge the children of `node` rather than having it access `node`'s parent and then merge the parent's children. I think making this change might simplify the function as you'd no longer have check if a parent exists.

Reply:
I had thought of doing that but `iterateFromLeaves` exits when it encounters a `SourceNode`. To implement your suggestion that behavior would have to be changed, so I took the less invasive way.

The check actually isn't necessary since it will encounter a `SourceNode` before getting an `undefined` parent and exit but I thought adding it wouldn't hurt.